api/v1/ptpoperatorconfig_webhook.go

```diff
@@ -0,0 +1,104 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"errors"
+	"net/url"
+
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var ptpoperatorconfiglog = logf.Log.WithName("ptpoperatorconfig-resource")
+
+var k8sclient client.Client
+
+func (r *PtpOperatorConfig) SetupWebhookWithManager(mgr ctrl.Manager, client client.Client) error {
+	k8sclient = client
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-ptp-openshift-io-v1-ptpoperatorconfig,mutating=false,failurePolicy=fail,sideEffects=None,groups=ptp.openshift.io,resources=ptpoperatorconfigs,verbs=update,versions=v1,name=vptpoperatorconfig.kb.io,admissionReviewVersions=v1
+
+const (
+	AmqScheme = "amqp"
+	// storageTypeEmptyDir is used for developer tests to map pubsubstore volume to emptyDir
+	storageTypeEmptyDir = "emptyDir"
+)
+
+func (r *PtpOperatorConfig) validate() error {
+	eventConfig := r.Spec.EventConfig
+	if eventConfig.EnableEventPublisher {
+		transportUrl, err := url.Parse(eventConfig.TransportHost)
+		if err == nil && transportUrl.Scheme == AmqScheme {
+			return nil
+		}
+		if eventConfig.StorageType == "" {
+			return errors.New("for HTTP transport, ptpEventConfig.storageType must be set to the name of StorageClass providing persist storage")
+		}
+		if eventConfig.StorageType != storageTypeEmptyDir && !r.checkStorageClass(eventConfig.StorageType) {
+			return errors.New("ptpEventConfig.storageType is set to StorageClass " + eventConfig.StorageType + " which does not exist")
+		}
+	}
+	return nil
+}
+
+var _ webhook.Validator = &PtpOperatorConfig{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *PtpOperatorConfig) ValidateCreate() error {
+	ptpoperatorconfiglog.Info("validate create", "name", r.Name)
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *PtpOperatorConfig) ValidateUpdate(old runtime.Object) error {
+	ptpoperatorconfiglog.Info("validate update", "name", r.Name)
+	return r.validate()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *PtpOperatorConfig) ValidateDelete() error {
+	ptpoperatorconfiglog.Info("validate delete", "name", r.Name)
+	return nil
+}
+
+func (r *PtpOperatorConfig) checkStorageClass(scName string) bool {
+
+	scList := &storagev1.StorageClassList{}
+	opts := []client.ListOption{}
+	err := k8sclient.List(context.TODO(), scList, opts...)
+	if err != nil {
+		return false
+	}
+
+	for _, sc := range scList.Items {
```

Comment:
what happens to PTP operator if someone deletes this storage class

Reply:
nothing would happen since the PV will still bound to the PVC if PV is not deleted either.

Reply:
verified, thanks